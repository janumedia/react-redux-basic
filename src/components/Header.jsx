import React from 'react';
import menu from '../data/menu';
import './Header.scss';

export default class Header extends React.Component {
    //right now we do optimization as the DOM for this component not required any update
    //but we might to updated later
    shouldComponentUpdate(nextProps, nextState) {
        return false;
    }
    render() {
        return (
            <div className="Header">
                <ul>
                {
                    menu.list.map(item => <li key={item.path}><a href={item.path}>{item.name}</a></li>)
                }
                </ul>
            </div>
        )
    }
}