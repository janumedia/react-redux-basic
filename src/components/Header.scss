.Header {
    background-color: #ECEC;
    ul {
        list-style: none;
        padding: 0;
    }
    li {
        display: inline-block;
        a {
            padding: 8px;
            display: block;
            transition: color 0.6s background-color 0.6s;
            &:hover {
                color: #FFF;
                background-color: #EEE;
            }
            &:link, &:active {
                color: inherit;
            }
        }
    }
}