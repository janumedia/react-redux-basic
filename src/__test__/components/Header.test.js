import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../../components/Header';

describe('Header component', () => {
    let component;
    
    beforeEach(() => {
        component = shallow(<Header/>)
    })
    
    it('should render properly', () => {
        expect(component.type()).toEqual('div');
        expect(component.hasClass('Header')).toBe(true);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should disable shouldComponentUpdate if state changed', () => {
        const expectedText = component.text();
        //try update state
        component.setState({testState:null});
        expect(component.text()).toEqual(expectedText);
    });

    it('should disable shouldComponentUpdate if props changed', () => {
        const expectedText = component.text();
        //try update state
        component.setProps({testProp:null});
        expect(component.text()).toEqual(expectedText);
    });
})