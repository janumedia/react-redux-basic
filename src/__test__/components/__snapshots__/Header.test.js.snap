// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render properly 1`] = `
<div
  className="Header"
>
  <ul>
    <li
      key="/"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      key="about"
    >
      <a
        href="about"
      >
        About
      </a>
    </li>
    <li
      key="works"
    >
      <a
        href="works"
      >
        Works
      </a>
    </li>
  </ul>
</div>
`;
